File: Leet Code/Insert into a Binary Search Tree/answer.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        if(root == NULL) return new TreeNode(val);

        TreeNode* temp = root;

        while(true){
            if(val >= temp->val){
                if(temp->right != NULL) 
                    temp = temp->right;
                else{
                    temp->right = new TreeNode(val);
                    break;
                }
            }
            else{
                if(temp->left != NULL) 
                    temp = temp->left;
                else{
                    temp->left = new TreeNode(val);
                    break;
                }
            }
        }

        return root;
    }
};

File: Leet Code/Increasing Order Search Tree/answer.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    TreeNode ans = new TreeNode(-1);
    public void solve(TreeNode root){
        if(root == null){
            return;
        }
        solve(root.left);
        root.left = null;
        ans.right = root;
        ans = root;
        solve(root.right);
    }
    public TreeNode increasingBST(TreeNode root) {
        TreeNode curr = ans;
        solve(root);
        return curr.right;
    }
}
